scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala

```diff
@@ -237,9 +237,9 @@ class PairSCollectionFunctions[K, V](val self: SCollection[(K, V)])
    */
   def hashJoin[W: ClassTag](that: SCollection[(K, W)])
   : SCollection[(K, (V, W))] = self.transform { in =>
-    val side = that.asMultiMapSideInput
+    val side = that.map(kv => Map(kv._1 -> Seq(kv._2))).sum.asSingletonSideInput
     in.withSideInputs(side).flatMap[(K, (V, W))] { (kv, s) =>
-      s(side).getOrElse(kv._1, Iterable()).iterator.map(w => (kv._1, (kv._2, w)))
+      s(side).getOrElse(kv._1, Seq.empty[W]).map(w => (kv._1, (kv._2, w)))
```

Comment:
using `asSingletonSideInput(Seq.empty[W])` did not work?

Reply:
hmm, the side input is a map, `getOrElse` is to return an empty seq for key that doesn't exist in the map.

Reply:
@daikeshi my bad - you are right.